_source/_posts/2022-02-07-cookies-vs-tokens.adoc

```diff
@@ -0,0 +1,135 @@
+
+Generally, the process works like this:
+
+- You sign in.
+- The server verifies your sign-in details and assigns you an authentication token.
+- The authentication token is used to make a request to your homepage that displays your unique dashboard.
+
+Both session cookies and access tokens allow users to make requests to the server without needing to re-authenticate at each request. Following is a comparison of the two.
+
+== What you should know about cookies
+
+Session cookies are stateful elements. They contain data that the server sends to the browser for temporary use. The authentication data inside a cookie is stored on both the client and server. The server keeps track of active sessions in a database, while the browser holds the identifier to the active session. When a request is made to the server, the session ID is used to look up user information such as their roles or privileges for authentication, in order to check if the session is still valid.
+
+=== Advantages of cookies
+
+**Cookies provide the following benefits. They:**
+
+* **Use the same session across subdomains**: Cookies take a https://en.wikipedia.org/wiki/HTTP_cookie#Domain_and_Path[`Domain`] argument: You specify the domain name for which the cookie is valid. Setting the domain name to `yoursite.com` allows the same session for the domain and subdomains.
+
+* **Reduce manipulation by client-side JavaScript**: Cookies allow you to restrict client-side access by setting the https://owasp.org/www-community/HttpOnly[`HttpOnly`] flag. This reduces the likelihood of cross-site scripting (XSS) attacks on your application, since most XSS attacks involve the use of malicious JS code.
+
+* **Require little storage**: Cookies use as little as 6 KB to store a simple user ID. Depending on what information you store in your cookie, you'll transmit a minimal amount of bytes with every request.
+
+* **Are managed by the browser**: This is automatic, so you don't have to worry about it.
+
+=== The downside of cookies
+
+**Some of the disadvantages of cookies include:**
+
+* ** Cross-site request forgery attacks (XSRF or CSRF)**: CSRF attacks are only possible with cookie-based session handling. The https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite[`SameSite`] attribute allows you to decide whether cookies should be sent to third-party apps using the **Strict** or **Lax** settings. A strict setting can prevent CSRF attacks, but it can also contribute to a poor browser experience for the user. For example, say your site uses a cookie named `tutorials_shown` to determine whether a user has already seen specific tutorials, in order to show them new tutorials every time they visit your site. If `SameSite` is set to `Strict` and someone follows a link to your site, the cookie will not be sent on that first request, and previously viewed tutorials will be shown. This creates a less personalized user experience.
```

Comment:
```suggestion
* ** Cross-site request forgery attacks (XSRF or CSRF)**: CSRF attacks are only possible with cookie-based session handling. The https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite[`SameSite`] attribute allows you to decide whether cookies should be sent to third-party apps using the **Strict** or **Lax** settings. A strict setting can prevent CSRF attacks, but it can also contribute to a poor browser experience for the user. For example, say your site uses a cookie named `tutorials_shown` to determine whether a user has already seen specific tutorials, in order to show them new tutorials every time they visit your site. If `SameSite` is set to `Strict` and someone follows a link to your site, the cookie will not be sent on that first request, and previously viewed tutorials will be shown. This creates a less personalized user experience.
```
```suggestion
* ** Cross-site request forgery attacks (XSRF or CSRF)**: CSRF attacks are only possible with cookie-based session handling. The https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite[`SameSite`] attribute allows you to decide whether cookies should be sent to third-party apps using the **Strict** or **Lax** settings. A strict setting can prevent CSRF attacks, but it can also contribute to a poor browser experience for the user. For example, say your site uses a cookie named `tutorials_shown` to determine whether a user has already seen specific tutorials in order to show them new ones every time they visit. If `SameSite` is set to `Strict` and someone follows a link to your site, the cookie will not be sent on that first request, and previously viewed tutorials will be shown. This creates a less personalized user experience.
```
clarifying